### ISSUE TITLE
Examples give X error: BadValue

### ISSUE THREAD

**Issue by pwaller** [azul3d/examples#10]
I'm just playing with things, and I thought I'd start off by building the examples on go1.5 (and subsequently I tried 1.4.2, too).

I can't get the examples to work at all though. I'm on Ubuntu 14.04, but I've also tried building with 15.04 with the same result.

```
$ go install -v azul3d.org/examples.v1/...
$ azul3d_triangle 
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  155 (GLX)
  Minor opcode of failed request:  16 (X_GLXVendorPrivate)
  Value in failed request:  0x340000a
  Serial number of failed request:  189
  Current serial number in output stream:  190
```

```
$ glxinfo
name of display: :0
display: :0  screen: 0
direct rendering: No (If you want to find out why, try setting LIBGL_DEBUG=verbose)
server glx vendor string: NVIDIA Corporation
server glx version string: 1.4
client glx vendor string: NVIDIA Corporation
client glx version string: 1.4
GLX version: 1.4
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: GeForce GTX 560 Ti/PCIe/SSE2
OpenGL version string: 2.1.2 NVIDIA 331.113
OpenGL shading language version string: 1.20 NVIDIA via Cg compiler
```

I've succeeded at getting `github.com/go-gl/examples/glfw31-gl21-cube` to run, but it runs extremely slowly (takes several seconds to start, then achieves a few FPS) and causes the window manager to freeze for seconds at a time. `github.com/go-gl/examples/glfw31-gl41core-cube` gives (which I don't understand because I believe my GPU supports 4.1):

```
$ glfw31-gl41core-cube 
panic: VersionUnavailable: GLX: Failed to create context: BadAlloc (insufficient resources for operation)

goroutine 1 [running, locked to thread]:
main.main()
	/home/pwaller/.local/src/github.com/go-gl/examples/glfw31-gl41core-cube/cube.go:45 +0x297

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
```

**Comment by slimsag** [comment]
Hi @pwaller, I apologize for the very late response on this. It's very hard to say what the issue is here.

Your error report seems to suggest that this is not a problem with Azul3D itself but perhaps with GLFW or more likely, your system configuration.

1. Are you able to run other OpenGL applications (say, `glxgears`) on the same system normally?

2. What output is given by the `azul3d_info` example/tool?

3. Your `glxinfo` output seems to suggest that you only have OpenGL 2.1.2 available, not 4.1. This highlights it as a driver issue. It is possible that you have accidentally mixed mesa and NVidia drivers together on the system, so I would suggest removing the mesa and NVidia drivers entirely and reinstalling Nvidia ones from scratch.

**Comment by pwaller** [comment]
Hi! No worries. I'm grateful for the free help.

Okay. I have no idea what was going on when I filed this report, but the situation has changed since I tried just now. Everything works and my glxinfo output is more sensible:

```
$ glxinfo | grep 'string'
server glx vendor string: NVIDIA Corporation
server glx version string: 1.4
client glx vendor string: NVIDIA Corporation
client glx version string: 1.4
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: GeForce GTX 560 Ti/PCIe/SSE2
OpenGL core profile version string: 4.3.0 NVIDIA 340.76
OpenGL core profile shading language version string: 4.30 NVIDIA via Cg compiler
OpenGL version string: 4.4.0 NVIDIA 340.76
OpenGL shading language version string: 4.40 NVIDIA via Cg compiler
```

Thanks for the great diagnostic suggestions! I can only assume something was temporarily very wrong with my environment. It's weird, because I don't recall doing anything which could cause that, so there must have just been something inconsistent about my computers state at that time. Perhaps Ubuntu had just done some updates and I needed to restart my session or something along those lines.
